desktop-widgets/profilewidget.cpp

```diff
@@ -338,25 +364,36 @@ void ProfileWidget::stopAdded()
 {
 	if (!editedDive)
 		return;
-	calcDepth(*editedDive, editedDc);
+	calcDepth(*editedDive, dc);
 	Setter s(placingCommand, true);
-	Command::editProfile(editedDive.get(), editedDc, Command::EditProfileType::ADD, 0);
+	Command::editProfile(editedDive.get(), dc, Command::EditProfileType::ADD, 0);
 }
 
 void ProfileWidget::stopRemoved(int count)
 {
 	if (!editedDive)
 		return;
-	calcDepth(*editedDive, editedDc);
+	calcDepth(*editedDive, dc);
 	Setter s(placingCommand, true);
-	Command::editProfile(editedDive.get(), editedDc, Command::EditProfileType::REMOVE, count);
+	Command::editProfile(editedDive.get(), dc, Command::EditProfileType::REMOVE, count);
 }
 
 void ProfileWidget::stopMoved(int count)
 {
 	if (!editedDive)
 		return;
-	calcDepth(*editedDive, editedDc);
+	calcDepth(*editedDive, dc);
+	Setter s(placingCommand, true);
+	Command::editProfile(editedDive.get(), dc, Command::EditProfileType::MOVE, count);
+}
+
+void ProfileWidget::stopEdited()
+{
+	if (!editedDive)
+		return;
+
+	copy_events(get_dive_dc(editedDive.get(), dc), get_dive_dc(d, dc));
```

Comment:
Revsisiting this, since I'm rebasing my C++ branch.

Does this work properly with undo? After all it seems that this overwrites the current dive outside of the undo system. Do you want me to look into that?

Reply:
Thanks for pointing this out. I had wondered why undo did not work, but it did not occur to me that I simply had to remove the `copy_events()` that I'd added before I switched to using the undo commands - doing this now in #4230, which works as it should, but feel free to include it in your changes if this makes it easier for you.